satpy/readers/yaml_reader.py

```diff
@@ -392,8 +392,10 @@ def new_filehandler_instances(self, filetype_info, filename_items):
             except KeyError:
                 logger.warning("Missing requirements for %s", filename)
                 continue
-
-            yield filetype_cls(filename, filename_info, filetype_info, *req_fh)
+            if reader_kwargs is None:
```

Comment:
I think it would be cleaner if at the top of this method you did:

```python
if reader_kwargs is None:
    reader_kwargs = {}
```

and then did `**reader_kwargs` regardless (getting rid of this `if` statement).

Reply:
done